check/configuration/custom-linters/reference.md

```diff
@@ -0,0 +1,527 @@
+
+[//]: # (`do_not_recommend_linters`: list of linters to not recommend &#40;actively disabled in repo&#41;)
+
+### `compile_commands`
+
+`compile_commands`: compile commands for clang-tidy. Must be one of `compile_commands_json` or `bazel`
+
+### `bazel`
+
+`bazel`: bazel configuration
+
+* `paths` locations to look for Bazel binary. [Example](../configuring-existing-linters/README.md#using-bazel)
+
+### `downloads`
+
+`downloads`: Locations to download binary artifacts from. Using [tool definitions](../../advanced-setup/tools/README.md) instead is preferred.
+
+[//]: # (`environments`: environment configuration)
+
+[//]: # (`triggers`: ??)
+
+### `comment_formats`
+
+`comment_formats`: definitions of comment formats. Reused in linter definitions.
+
+### `exported_configs`
+
+`exported_configs`: Linter configs to export when another project is [importing this plugin](../sharing-linters.md) 
+
+[//]: # (`shared_configs`: ???)
+
+### `default_max_file_size`
+
+`default_max_file_size`: Default maximum filesize in bytes. Trunk Check will not run linters on any files larger than this. Default value is 4 megabytes.
+
+### `extra_compilation_flags`
+
+`extra_compilation_flags`: When running clang-tidy, this list will be appended to the compile command.
+
+### `compile_commands_roots`
+
+`compile_commands_roots`: Directories to search for `compile_commands.json`.
+
+[//]: # (`hold_the_line_mode`: which algorithm to use for hold the line)
+
+
+## Linter Definition
+
+The definition of a particular linter is put under `lint.definitions`.  The following
+properties define the settings of a *particular linter*, not for all linters. For
+global linter settings, see [Lint Config](reference.md#lint-config)
+
+
+### `name`
+
+`name` Is the name of the linter. This property will be used to refer to the linter in other 
+parts of the config, for example, in the list of enabled linters.
+
+[//]: # (`type` is the type of output the linter produces. **link to output types**)
+
+[//]: # (`command`, string, command to run: ex: `[$workspace/tools/linters/clang-format, --assume-filename=${path}]`)
+
+[//]: # (`success_codes` is the list of success codes that indicates linter generated data and should be processed. Some linters return 0 for nothing to lint and 1 when there are issues.)
+
+[//]: # (`error_codes` is the list of error codes this linter will return when it hit an internal failure and couldn't generate results.)
+
+### `direct_configs`
+
+__If you have one of these config files, you're definitely using this linter__
+
+### `affects_cache`
+
+`affects_cache`: the list of files that affect the cache results of this linter.
+
+### `good_without_config`
+
+`good_without_config`: *optional boolean*. Indicates whether this linter is recommended without 
+the user tuning its configuration. Prefer `suggest_if`.
+
+### `files`
+
+`files` is a list of file types listed in the `lint.files` section that this linter applies to. 
```

Comment:
Delete and move this text to the commands section files. The files description there is incorrect.